odoo/addons/base/models/ir_ui_view.py

```diff
@@ -431,13 +431,34 @@ def _valid_inheritance(self, arch):
                         self._raise_view_error(message, node)
         return True
 
+    def _validate_render_report(self):
+        for view in self:
+            if view.type != 'qweb':
+                return
+            arch_value = 't-call="%s"' % view.xml_id
+            parent_views = self.env['ir.ui.view'].search([('type', '=', 'qweb'), ('arch_db', 'ilike', arch_value)])
+            if parent_views:
+                parent_views._validate_render_report()
+            report = self.env['ir.actions.report'].search([('report_name', '=', view.xml_id)], limit=1)
+            if len(report) < 1:
+                return
+            try:
+                report_data = self.env[report.model].search([], limit=1)
+                if len(report_data) < 1:
+                    return
+                context = dict(self.env.context)
+                report.with_context(context)._render_qweb_html(report.report_name, [report_data.id])
+            except Exception as e:
+                raise ValidationError(_(f"Error while rendering view: \n\n{e}"))
+
     @api.constrains('arch_db')
     def _check_xml(self):
         # Sanity checks: the view should not break anything upon rendering!
         # Any exception raised below will cause a transaction rollback.
         partial_validation = self.env.context.get('ir_ui_view_partial_validation')
         self = self.with_context(validate_view_ids=(self._ids if partial_validation else True))
 
+        self._validate_render_report()
```

Comment:
you should probably move this call a bit lower, in the `if view.type == 'qweb':` bloc

you could then pbly remove the try/except bloc in your method, as the errors would be caught in the try/except bloc around this one